zk_toolbox/crates/zk_inception/src/commands/prover/run.rs

```diff
@@ -29,114 +30,114 @@ pub(crate) async fn run(args: ProverRunArgs, shell: &Shell) -> anyhow::Result<()
+    };
+
+    if in_docker {
+        let path_to_configs = chain.configs.clone();
+        let path_to_prover = get_link_to_prover(&ecosystem_config);
+        run_dockerized_component(
+            shell,
+            component.image_name(),
+            &application_args,
+            &additional_args,
+            message,
+            error,
+            &path_to_configs,
+            &path_to_prover,
+        )?
+    } else {
+        run_binary_component(
+            shell,
+            component.binary_name(),
+            &application_args,
+            &additional_args,
+            message,
+            error,
+        )?
     }
 
     Ok(())
 }
 
-fn run_gateway(shell: &Shell, chain: &ChainConfig) -> anyhow::Result<()> {
-    logger::info(MSG_RUNNING_PROVER_GATEWAY);
-    let config_path = chain.path_to_general_config();
-    let secrets_path = chain.path_to_secrets_config();
-
-    let mut cmd = Cmd::new(cmd!(shell, "cargo run --release --bin zksync_prover_fri_gateway -- --config-path={config_path} --secrets-path={secrets_path}"));
-    cmd = cmd.with_force_run();
-    cmd.run().context(MSG_RUNNING_PROVER_GATEWAY_ERR)
-}
-
-fn run_witness_generator(
+fn run_dockerized_component(
     shell: &Shell,
-    chain: &ChainConfig,
-    args: WitnessGeneratorArgs,
+    image_name: &str,
+    application_args: &[String],
+    args: &[String],
+    message: &'static str,
+    error: &'static str,
+    path_to_configs: &PathBuf,
+    path_to_prover: &PathBuf,
 ) -> anyhow::Result<()> {
-    logger::info(MSG_RUNNING_WITNESS_GENERATOR);
-    let config_path = chain.path_to_general_config();
-    let secrets_path = chain.path_to_secrets_config();
-    let round = args.round.expect(MSG_WITNESS_GENERATOR_ROUND_ERR);
+    let spinner = Spinner::new(&format!("Pulling image {}...", image_name));
+    let pull_cmd = Cmd::new(cmd!(shell, "docker pull {image_name}"));
+    pull_cmd.run().context(error)?;
+    spinner.finish();
 
-    let round_str = match round {
-        WitnessGeneratorRound::AllRounds => "--all_rounds",
-        WitnessGeneratorRound::BasicCircuits => "--round=basic_circuits",
-        WitnessGeneratorRound::LeafAggregation => "--round=leaf_aggregation",
-        WitnessGeneratorRound::NodeAggregation => "--round=node_aggregation",
-        WitnessGeneratorRound::RecursionTip => "--round=recursion_tip",
-        WitnessGeneratorRound::Scheduler => "--round=scheduler",
-    };
+    logger::info(message);
 
-    let mut cmd = Cmd::new(cmd!(shell, "cargo run --release --bin zksync_witness_generator -- {round_str} --config-path={config_path} --secrets-path={secrets_path}"));
-    cmd = cmd.with_force_run();
-    cmd.run().context(MSG_RUNNING_WITNESS_GENERATOR_ERR)
-}
+    let mut cmd = Cmd::new(cmd!(
+        shell,
+        "docker run --net=host -v {path_to_prover}/data/keys:/prover/data/keys -v {path_to_prover}/artifacts:/artifacts -v {path_to_configs}:/configs {application_args...} {image_name} {args...}"
+    ));
```

Comment:
minor: we can do it as a follow up later, but can we adapt our [docker::run](https://github.com/matter-labs/zksync-era/blob/main/zk_toolbox/crates/common/src/docker.rs#L20) for that? ignore, if that's too much work.

Reply:
btw it's pretty tricky. Initially I tried to put all the docker args in the `application_args` but it seems that xshell puts args with `:` in a double quotes, which breaks the command.